### PR TITLE
vagrantnode.go: improve resiliency by re-dialing on session create error

### PR DESCRIPTION
**Pull request by erikh**
This fixes a number of reliability issues with `ssh.Session`s. Instead, it now dials each time, which is technically slower but in practice isn't a huge Issue.

Note that for mega-parallel tests you may wish to create, you'll need to adjust the `MaxStartups` and `MaxSessions` variables will require large increases.

I'll be updating the build repository soonish.

### CODE REVIEW

File: vagrantnode.go [hunk 1]
```diff
@@ -113,15 +120,10 @@ func (n *VagrantNode) RunCommandWithOutput(cmd string) (string, error) {
...
-		s   *ssh.Session
-		err error
-	)
-
-	if s, err = n.client.NewSession(); err != nil {
+	_, s, err := n.GetClientAndSession()
+	if err != nil {
 		return err
 	}
-	defer s.Close()
```
Comment (mapuri) [inline vagrantnode.go:124]: just curious, should we be doing Close for client and session here as well?

**Comment by mapuri** [comment]
LGTM

so comparing this approach to doing `vagrant ssh`, I think this is still better as it doesn't have the overhead of starting a new process?

File: vagrantnode.go (hunk 1 above)
Comment (erikh) [inline vagrantnode.go:124]: that would eliminate the process, which we don't want. Instead we leak a connection and wait for it to clean up itself.

I'm not sure there's a better solution.

**Comment by erikh** [comment]
The overhead of a ruby process is very significant, measured in seconds.

**Comment by mapuri** [comment]
yeah, that's what I thought.

LGTM, I will merge this.

File: vagrantnode.go (hunk 1 above)
Comment (mapuri) [inline vagrantnode.go:124]: shall we put a comment or something here?

Comment (erikh) [inline vagrantnode.go:124]: good idea. don't merge yet.

Comment (mapuri) [inline vagrantnode.go:124]: sure, please feel free to merge once done.

**Comment by erikh** [comment]
ok, PTAL

**Comment by mapuri** [comment]
looks good

**Pull request merged**